builder/builder-main.c

```diff
@@ -74,6 +75,7 @@ static GOptionEntry entries[] = {
   { "disable-download", 0, 0, G_OPTION_ARG_NONE, &opt_disable_download, "Don't download any new sources", NULL },
   { "disable-updates", 0, 0, G_OPTION_ARG_NONE, &opt_disable_updates, "Only download missing sources, never update to latest vcs version", NULL },
   { "download-only", 0, 0, G_OPTION_ARG_NONE, &opt_download_only, "Only download sources, don't build", NULL },
+  { "use-sources", 0, 0, G_OPTION_ARG_STRING_ARRAY, &opt_sources_dirs, "Specify a directory with sources to build from", "SOURCE-DIR"},
```

Comment:
Also, some info about sources dir precendence would be useful. From the code I can say that the directories from the `--use-sources` flag have precendence over the ones in manifest. Also the directories are checked in the same order as specified, i.e. the first `--use-sources` directory has precendence over the rest.